final-project/application.py

```diff
@@ -519,183 +519,98 @@ def getProductJSON():
-        # the intended name.
-        # e.g. while (request.form.get('reference-counterNumber')) {
-            # make_edits['reference']['title'] = request.form.getList['reference']['title']
-            # make_edits['reference']['link'] = request.form.getList['reference']['link']
-            # counterNumber++
-        # }
-        # This loop will end when there is no longer a form with the name we requested.
-        # TODO: Check whether .getList['reference']['title'] is the correct syntax.
-        # Essentially, will getList intelligently store multiple inputs from a single form
-        # into a dict.
-        # It could attempt to store them into a list or some other such nonsense.
 
 
-    else:
-        # the code below is executed if the request method
-        # was GET or the credentials were invalid
-        # TODO:
-        # Remove render_template.
-        # Return successful information edit message
-        # Call function to return new product information.
-        # New product information can be called here, or in javascript.
-        # Javascript would be better to keep this a single use funciton.
-        # Remove error varilable from argument list.
-        return render_template('productJSON.html', error=error)
 
-    # TODO:
-    # Ensure blank inputs are not accepted.
-    # Loop through make_edits dict to check for odd information.
+        # PART 3: Return the new product information.
 
+        # Store the product id..
+        product_id_of_the_request = prod_data['product-id']
 
+        # Get the updated information for this product id.
+        product_info = find_product(product_id_of_the_request)
 
+        # If no product was found, then apologize.
+        if not product_info or product_info == None:
+            return apology("Sorry, we didn't find a product with id "
+                + str(product_id_of_the_request))
 
-    # Find our record of this product
-    product_info = find_product(id_products_request)
+        # Get the references.
+        (number_of_references,
+        reference_titles,
+        reference_ids,
+        reference_links) = get_reference(product_info[0]["product_name"])
 
-    # Get references for a given product
-    (number_of_references,
-    reference_titles,
-    reference_links) = get_reference(id_products_request)
-
-    # If no product was found, then apologize.
-    if not product_info or product_info == None:
-        return apology("Sorry, we don't have information for that product")
-    elif len(product_info) > 1:
-        return apology("Found more than a single product. Please try again.")
+        # Append the references to our product information.
+        product_info[0]['number_of_references'] = number_of_references
+        product_info[0]['reference_ids'] = reference_ids
+        product_info[0]['reference_titles'] = reference_titles
+        product_info[0]['reference_links'] = reference_links
```

Comment:
Not sure why this needs to be in an array, since you are just accessing the first element.